### PR TITLE
Epic: Onboarding for 3.0

### PR DESCRIPTION
**Pull request by alaca**
## Description
This PR adds the Admin onboarding UI for the new Form Builder. 

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

**Comment by kjohnson** [comment]
@alaca have we hooked up the "Test it out" button?
image

**Comment by alaca** [comment]
@kjohnson the "Test it out" button is now working https://github.com/impress-org/givewp/pull/6832/commits/4833cc9a9346c281b8815719573328ef52f087f1

**Comment by kjohnson** [comment]
@alaca this looks good.

We need to consider our release strategy here - being a branch of the core GiveWP repo we can't just merge it into `develop` because we aren't ready for this to be public yet. Could we work in a feature flag that checks for Next Gen (temporarily) or migrate this to the Next Gen plugin and extend the table with slots?

**Comment by alaca** [comment]
@kjohnson good point. I'd rather go with the flag route, merging this into NextGen will require a lot of rewriting and we don't have the capacity for that right now. Also, the ListTable component that we use to display donation forms is built in such a way that is not easy to hook up to it.

File: src/DonationForms/V2/DonationFormsAdminPage.php [hunk 1]
```diff
@@ -202,4 +218,34 @@ public static function getUrl()
...
+
+    /**
+     * Get an array of unsupported addons for addons compatibility modal
+     *
+     * @unreleased
+     * @return array
+     */
+    public function getUnsupportedAddons(): array
+    {
+        $unsupportedAddons = [
```
Comment (kjohnson) [inline src/DonationForms/V2/DonationFormsAdminPage.php:230]: @alaca We need to fill out this unsupported list before we merge.

Comment (alaca) [reply src/DonationForms/V2/DonationFormsAdminPage.php:230]: @kjohnson I'm not sure which ones are supported or not, so I will add them all for now.

Comment (alaca) [reply src/DonationForms/V2/DonationFormsAdminPage.php:230]: List updated https://github.com/impress-org/givewp/pull/6832/commits/c3e912ca9f61d7481de447d067dc8cbd46ac0228

File: src/DonationForms/V2/Endpoints/ListDonationForms.php [hunk 2]
```diff
@@ -128,10 +128,21 @@ public function handleRequest(WP_REST_Request $request): WP_REST_Response
             $this->listTable->items($forms, $this->request->get_param('locale') ?? '');
             $items = $this->listTable->getItems();
 
-            foreach($items as &$item ) {
+            foreach ($items as &$item) {
+                $v2form = defined('GIVE_NEXT_GEN_VERSION') && ! get_post_meta($item['id'], 'formBuilderFields');
+                $migrate = $v2form
+                           && ! get_post_meta($item['id'], 'migratedFormId');
```
Comment (kjohnson) [inline src/DonationForms/V2/Endpoints/ListDonationForms.php:134]: This _should_ be checking if the v2 form ID is specified as a `migratedFormId` on a v3 form. The v3 form holds the reference to a migrated v2 form.

File: src/DonationForms/V2/Endpoints/ListDonationForms.php [hunk 3]
```diff
@@ -128,10 +128,21 @@ public function handleRequest(WP_REST_Request $request): WP_REST_Response
...
+
+                $transfer = $v2form
+                            && get_post_meta($item['id'], 'migratedFormId')
+                            && ! get_post_meta($item['id'], 'transferredFormId');
```
Comment (kjohnson) [inline src/DonationForms/V2/Endpoints/ListDonationForms.php:138]: Similar to the `migratedFormId` the reference for the `transferredFormId` is stored on a corresponding v3 form.

**Comment by kjohnson** [comment]
@alaca locally I'm trying our `give_is_form_migrated($formId)` and `give_is_form_donations_transferred($formId)` for the v2 form checks (against v3 forms). I'm not sure if these should be a part of Next Gen or GiveWP core. That said, they will eventually be combined, so it probably isn't that big of a deal.

File: src/DonationForms/V2/Endpoints/ListDonationForms.php (hunk 2 above)
Comment (alaca) [reply src/DonationForms/V2/Endpoints/ListDonationForms.php:134]: @kjohnson it makes sense that the v3 form holds the reference to the migrated v2 form, but still, we need some flag on the v2 form in order to show/hide migrate and transfer links. Without that, I will have to loop through all v3 forms and collect references from them, and then loop again through all v2 forms and check if the v2 form id exists in the collection. This is not reliable. I suggest that we add a flag on the v2 form as well.

**Comment by kjohnson** [comment]
Here is an example:

```php
function give_is_form_migrated($formId) {
    global $wpdb;

    return (bool) DB::get_var(
        DB::prepare(
            "
                    SELECT `form_id`
                    FROM `{$wpdb->prefix}give_formmeta`
                    WHERE `meta_key` = 'migratedFormId'
                      AND `meta_value` = %d",
            $formId
        )
    );
}
```

**Comment by alaca** [comment]
The trashed v3 form will still have all the metadata, therefore no we don't have to check anything extra. If the v3 form is deleted, the metadata will be deleted as well.

**Comment by kjohnson** [comment]
> If the v3 form is deleted, the metadata will be deleted as well.

Except that the to be deleted form is set as "trashed" for 30 days before being fully deleted.
That and the new UI does not support force deleting forms.

I added a join to the query to account for "not trashed" status.

**Comment by kjohnson** [comment]
> the new UI does not support force deleting forms.

The work around is to "trash" the v3 form in the new UI, then switch to the old UI, view "trashed" forms and then permanently delete the form, then switch back to the new UI.

**Comment by kjohnson** [comment]
As a note, we are putting a small hold on this to get feedback from the product owners.

**Comment by alaca** [comment]
@kjohnson can you check why tests are failing? Also, I'm not able to generate the zip file for QA

**Comment by kjohnson** [comment]
Closing in favor of https://github.com/impress-org/givewp/pull/6914.

I messed up a merge which caused some issue with development in other branches. @alaca was kind enough to port the intended changes from this branch to a new branch to fix my mistake.

**Pull request closed without merging**